src/components/SlateEditor/plugins/PluginFactory.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2024-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { Editor, NodeEntry, Element } from "slate";
+import { RenderLeafProps } from "slate-react";
+import config from "../../../config";
+import { NormalizerConfig, defaultBlockNormalizer } from "../utils/defaultNormalizer";
+
+type ElementType = Element["type"];
+
+type MappedPlugins = {
+  [K in ElementType]: Plugin<K>;
+};
+
+interface Plugin<T extends ElementType = ElementType> {
+  type: T;
+  normalize?: Normalize<Extract<Element, { type: T }>>[];
+  normalizerConfig?: NormalizerConfig;
```

Comment:
Burde forskjellen på `normalize` og `normalizerConfig` vært tydeligere i navngivningen? Jeg syntes ikke det er intuitivt at normalize er spesifikk normaliserings-logikk for en plugin mens normalizerConfig er bruk av default block normalizer ..